cmd/snap/cmd_services.go

```diff
@@ -184,8 +185,63 @@ func (s *svcLogs) Execute(args []string) error {
 	return nil
 }
 
+type userAndScopeMixin struct {
+	System bool   `long:"system"`
+	User   bool   `long:"user"`
+	Users  string `long:"users"`
+}
+
+var userAndScopeDescs = mixinDescs{
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"system": i18n.G("The operation should only affect system services."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"user": i18n.G("The operation should only affect user services for the current user."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"users": i18n.G("The operation should affect all user services."),
+}
+
+func (um *userAndScopeMixin) validateScopes() error {
+	switch {
+	case um.System && um.User:
+		return fmt.Errorf("--system and --user cannot be used in conjunction with each other")
+	case um.Users != "" && um.User:
+		return fmt.Errorf("--user and --users=all cannot be used in conjunction with each other")
```

Comment:
maybe swap the case order or drop `all` from the error message as at this point we have not yet verified that `--users=all` was indeed provided

```suggestion
		return fmt.Errorf("--user and --users cannot be used in conjunction with each other")
```